components/brave_wallet_ui/components/buy-send-swap/tabs/swap-tab.tsx

```diff
@@ -120,7 +122,7 @@ function SwapTab (props: Props) {
   }
 
   const onFilterAssetList = (asset: BraveWallet.BlockchainToken) => {
-    const newList = assetOptions.filter((assets) => assets !== asset)
+    const newList = swapToListAssetOptions.filter((assets) => assets !== asset)
```

Comment:
`swapToListAssetOptions` vs `swapFromListAssetOptions` should be selected based on `isSelectingAsset`. I think this will fix the search issue you're facing.